### PR TITLE
Add Qt.py

### PR DESCRIPTION
**Pull request by fredrikaverpil**
> Minimal Python 2 & 3 shim around all Qt bindings - PySide, PySide2, PyQt4 and PyQt5.

I'm one of the maintainers of this project and I wish to add it to Conda. More info & docs over at the project's website: https://github.com/mottosso/Qt.py

### CODE REVIEW

**Comment by fredrikaverpil** [comment]
@conda-forge/core can someone restart the AppVeyor build for this PR?
Seems to me to be the only thing left before it could be merged.

screen shot 2017-08-27 at 20 35 37

**Comment by ocefpaf** [comment]
You are listed as a maintainer, correct? You should have rights to re-start AppVeyor. (If not closing and re-opening also works, but that would trigger all the CIs... Still better than waiting for someone else to re-start  it :wink:)

**Comment by fredrikaverpil** [comment]
> You are listed as a maintainer, correct? 

Yes!

> You should have rights to re-start AppVeyor. 

Unless I'm mistaken (see the screenshot I posted), I cannot restart it.

I see you restarted it. Thanks!

**Comment by fredrikaverpil** [comment]
@conda-forge/core I don't know if AppVeyor is having issues today or something, but this is ready to be merged.

**Comment by ocefpaf** [comment]
> Unless I'm mistaken (see the screenshot I posted), I cannot restart it.

Sorry, read this on my phone and did not see that it is staged-recipes. You'll have access in the feedstock.

File: recipes/qt.py/existance_check.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        print('Qt.py was not found in any of the locations: ' + ', '.join(
+            site_package_locations))
+    return found
+
+
+if __name__ == '__main__':
+    if existance_check():
+        sys.exit(0)
+    else:
+        sys.exit(1)
```
Comment (ocefpaf) [inline recipes/qt.py/existance_check.py]: This looks like an overly complex script for a simple test :smile: 
Are there any `import` test that could substitute this?

Comment (fredrikaverpil) [reply recipes/qt.py/existance_check.py]: No, unfortunately, there isn't. Qt.py requires you to have PySide, PyQt4, PySide2 or PyQt5 installed and will raise an error if none of them are found. I can't add any of these bindings as requirements in `meta.yaml` as I wish the users of Qt.py to decide which of these bindings they want to use.

I'm not aware of any other way to tell if the `Qt.py` file is in place. Would you recommend some other way?

Comment (isuruf) [reply recipes/qt.py/existance_check.py]: Then add `PySide` as a test requirement
```
test:
    requires:
       - PySide
    imports:
       - Qt.py
```

File: recipes/qt.py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
+    - python
+
+test:
+  requires:
+    - PySide2
```
Comment (isuruf) [inline recipes/qt.py/meta.yaml]: this should be `pyside2`

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Hm, is PySide really available for Python > 3.4?

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: PySide2 was added as a feedstock just an hour ago or so. It should work much better I think. But it wasn't found on all platforms. Trying PySide now.

Comment (isuruf) [reply recipes/qt.py/meta.yaml]: I didn't mean `pyside`. I meant the package name should be lowercase. `pyside2` instead of `PySide2`

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Doesn't seem like PySide worked either: https://travis-ci.org/conda-forge/staged-recipes/builds/268999200#L278

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Trying PyQt4 then...

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Oh! I see now. It was case sensitive!

Comment (isuruf) [reply recipes/qt.py/meta.yaml]: Please see my comment here, https://github.com/conda-forge/staged-recipes/pull/3832#discussion_r135422099

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Yeah, now it works. Thanks @isuruf 😄

**Comment by fredrikaverpil** [comment]
@conda-forge/core still waiting for AppVeyor. Please merge at your will. 😄

**Comment by fredrikaverpil** [comment]
@conda-forge/core Could someone please help me merge this as it keeps failing with AppVeyor due to [current build issues they're having](https://appveyor.statuspage.io/)?

I use this project on a daily basis on all platforms, using multiple Python versions and I can vouch for that it'll work just fine as is.

**Comment by fredrikaverpil** [comment]
Closing in order to re-trigger CI.

**Comment by fredrikaverpil** [comment]
@ocefpaf @isuruf all checks ok. You can go ahead and merge this if you think it looks alright! 😄

File: recipes/qt.py/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+test:
+  requires:
+    - pyside2
+  imports:
+    - Qt
+
+about:
+  home: http://github.com/mottosso/Qt.py
+  license: MIT
+  license_family: MIT
```
Comment (isuruf) [inline recipes/qt.py/meta.yaml]: Can you add `license_file: LICENSE` here?

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Ok, done. Is the actual `LICENSE` file going to be generated?

Comment (isuruf) [reply recipes/qt.py/meta.yaml]: Adding this line would copy the file with the name `LICENSE` from the source folder to the conda package.

Comment (isuruf) [reply recipes/qt.py/meta.yaml]: See https://conda-forge.org/docs/meta.html#packaging-the-license-manually

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Looks like I need to add a `LICENSE` file. Is that really required for the MIT license?

Should I add a `LICENSE` next to `meta.yaml` and this should work?

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Ok, thanks!

File: recipes/qt.py/meta.yaml [hunk 4]
```diff
@@ -32,7 +32,7 @@ about:
   home: http://github.com/mottosso/Qt.py
   license: MIT
   license_family: MIT
-  license_file: LICENSE
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (isuruf) [inline recipes/qt.py/meta.yaml]: Thanks for adding this. Can you also raise the issue upstream about including the file in the source distribution? Adding a MANIFEST.IN should solve this.

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: Sure! Thanks @isuruf 😄

Comment (fredrikaverpil) [reply recipes/qt.py/meta.yaml]: @isuruf would it be fine to merge this without waiting for AppVeyor? I would really like to start using this as soon as possible.

**Pull request merged**